Adafruit_BNO055.h

```diff
@@ -279,7 +279,7 @@ class Adafruit_BNO055 : public Adafruit_Sensor
       VECTOR_GRAVITY       = BNO055_GRAVITY_DATA_X_LSB_ADDR
     } adafruit_vector_type_t;
 
-#ifdef ARDUINO_SAMD_ZERO
+#if defined (ARDUINO_SAMD_ZERO) && ! (ARDUINO_SAMD_FEATHER_M0)
```

Comment:
@aethaniel As far as I can tell, the problem the original code is defending against is to do with the additional EDBG (external debugging) chip used on the Zero, which uses the same I2C address as the BNO055 - not a problem with the SAMD architecture. So no it doesn't apply to derivatives unless they also have the same debug chip (which the feather does not).

I think it may be a problem that when compiling sketches for the the Feather, the  ARDUINO_SAMD_ZERO condition is true for the feather. From what your saying it would be better if the feather did not identify as a zero (even if that is probably pragmatic for many existing sketches?) and sketch writers were careful to use the ARDUINO_ARCH_SAMD or ARDUINO_SAMD_ZERO conditions correctly? (which I think this one does in fact)